ivy/functional/frontends/jax/numpy/linalg.py

```diff
@@ -152,3 +154,12 @@ def cond(x, p=None):
     nan_mask = ivy.logical_and(ivy.isnan(r), ~ivy.isnan(x).any(axis=(-2, -1)))
     r = ivy.where(orig_nan_check, ivy.where(nan_mask, ivy.inf, r), r)
     return r
+
+
+@to_ivy_arrays_and_back
+def multi_dot(arrays, /, *, precision=None):
+    dtype = arrays[0].dtype
+    for i in range(1, len(arrays)):
+        dtype = jax.numpy.promote_types(dtype, arrays[i].dtype)
+    arrays = list(array.astype(dtype) for array in arrays)
```

Comment:
We can avoid this explicit dtype casting since the other ivy frontends doesn't employ this.

Reply:
Sorry, I'm not sure exactly what you mean here. Without the type casting some of the tests where failing because the output had the wrong dtypes. Should I just admit the errors and accept the test failures?

Reply:
In that case, don't you think it should be present with other frontends as well? I think we should fix the issues that occurs when not type casting(if anything comes in). Let's see how it goes :slightly_smiling_face:

Reply:
Also it's not allowed to import jax to the frontend I guess. May be this caused other tests not related to your pr to fail.

Reply:
I would suggest you try locally and if errors still persists, you could use type casting but instead of jax, use some ivy function to do this.

Reply:
That does make sense, sorry I should have realised that earlier. Do you know if there are there any ivy functions that emulate a frontends' dtype promotion rules? I can't seem to find any

Reply:
I think there is something like `promote_type`. You could search the docs for something like this.